cws-engine-service/src/main/java/jpl/cws/engine/EngineDbService.java

```diff
@@ -144,64 +148,76 @@ public void updateStats(File[] logs, long diskFreeBytes) {
 	 */
 	public void createOrUpdateWorkerRow(String lockOwner) {
 		if (!workerExists()) {
-			//
-			// ROW FOR WORKER DOES NOT EXIST, SO CREATE ONE
-			//
+			if ( cwsInstallType.equals("console_only") && workerConsoleOnlyTypeExists() ) {
+				//
+				// ROW FOR CONSOLE ONLY WORKER ALREADY EXISTS
+				//
+				log.info("Worker Row with cws_install_type: " + cwsInstallType + ", already exists. ");
+			}
+			else {
+				//
+				// ROW FOR WORKER DOES NOT EXIST, SO CREATE ONE
+				//
 
-			int numUpdated = 0;
-			int numTries = 0;
-			String workerName = null;
-			while (numTries++ < 10 && numUpdated != 1) {
-				Timestamp tsNow = new Timestamp(DateTime.now().getMillis());
+				int numUpdated = 0;
+				int numTries = 0;
+				String workerName = null;
+				while (numTries++ < 10 && numUpdated != 1) {
+					Timestamp tsNow = new Timestamp(DateTime.now().getMillis());
 
-				// Determine a worker name, using current system time in milliseconds.
-				// TODO: In the future, we might want to do a more elegant algorithm,
-				//       but this may just be fine for the long-run
-				workerName = "worker-" + System.currentTimeMillis();
-				
-				try {
-					log.info("Inserting row (attempt #" + numTries +", workerName='"+workerName+"') into cws_worker table...");
+					// Determine a worker name, using current system time in milliseconds.
+					// TODO: In the future, we might want to do a more elegant algorithm,
+					//       but this may just be fine for the long-run
+					workerName = "worker-" + System.currentTimeMillis();
+					// newly added
+					if (cwsInstallType.equals("console_only")) {
+						workerName = "worker-" + "0000";
```

Comment:
As per #54, this should be 

```suggestion
						workerName = "worker0000";
```